Aquifer/Grouping.swift

```diff
@@ -8,36 +8,14 @@
 
 // roughly `Pipes.Group`
 
-import Foundation
 import Swiftz
 
-internal enum GroupedProducerRepr<V, R> {
-    case End(() -> R)
-    case More(() -> Proxy<X, (), (), V, GroupedProducerRepr<V, R>>)
-
-    internal func fmap<N>(f: R -> N) -> GroupedProducerRepr<V, N> {
-        switch self {
-        case let .End(x): return GroupedProducerRepr<V, N>.End { _ in f(x()) }
-        case let .More(p): return GroupedProducerRepr<V, N>.More { _ in { q in q.fmap(f) } <^> p() }
-        }
-    }
-
-    internal func ap<N>(f: GroupedProducerRepr<V, R -> N>) -> GroupedProducerRepr<V, N> {
-        switch (self, f) {
-        case let (.End(x), .End(g)): return GroupedProducerRepr<V, N>.End { _ in g()(x()) }
-        case let (.More(p), .End(g)): return GroupedProducerRepr<V, N>.More { _ in p().fmap { q in q.fmap(g()) } }
-        case let (_, .More(g)): return GroupedProducerRepr<V, N>.More { _ in g().fmap { h in self.ap(h) } }
-        }
-    }
-
-    internal func bind<N>(f: R -> GroupedProducerRepr<V, N>) -> GroupedProducerRepr<V, N> {
-        switch self {
-        case let .End(x): return f(x())
-        case let .More(p): return GroupedProducerRepr<V, N>.More { _ in p().fmap { q in q.bind(f) } }
-        }
-    }
-}
-
+/// A producer that uses the Free Monad Transformer (unrepresentable in Swift) to delimit groupings
+/// of produced values.
+///
+/// Ideally:
+///
+///    typealias GroupedProducer<V, R> = Free<Producer<V> R>
```

Comment:
Should probably be `Free<Producer<V, $0>, R>` just for consistency with Swift syntax.  :)